contracts/stellar-interchain-token/src/contract.rs

```diff
@@ -125,13 +125,25 @@ impl InterchainTokenInterface for InterchainToken {
 
     #[only_owner]
     fn add_minter(env: &Env, minter: Address) {
+        assert_with_error!(
+            env,
+            !Self::is_minter(env, minter.clone()),
+            ContractError::MinterAlreadyExists
+        );
+
```

Comment:
Look at where `add_minter` is being used. The ITS logic relies adding the same minter being a no-op. If we restrict it here, then that logic should be updated to handle not cause a failure there